### PR TITLE
Create embedded config on first use

### PR DESCRIPTION
**Pull request by sbc100**
When emscripten is first used and we create the default config file,
create it in the emscripten directory if we can.

### CODE REVIEW

File: ChangeLog.md [hunk 1]
```diff
@@ -17,6 +17,11 @@ See docs/process.md for how version tagging works.
 
 Current Trunk
 -------------
+- On first use, emscripten creates a sample config file.  This config file
+  is now created in the emscripten directory by deafult.  The traditional
```
Comment (kripken) [inline ChangeLog.md:21]: typo in `default`

File: docs/emcc.txt [hunk 2]
```diff
@@ -429,9 +429,9 @@ Options that are modified or new in *emcc* are listed below:
    If using this in combination with "--clear-cache", be sure to
    specify this argument first.
 
-   The Emscripten cache defaults to being located in the path name
-   stored in the "EM_CACHE" environment variable or
-   "~/.emscripten_cache".
+   The Emscripten cache defaults "emscripten/cache" but can be
```
Comment (kripken) [inline docs/emcc.txt:432]: missing `to` after `defaults`

File: docs/emcc.txt [hunk 3]
```diff
@@ -429,9 +429,9 @@ Options that are modified or new in *emcc* are listed below:
...
+   overridden use "EM_CACHE" environment variable or "CACHE" config
```
Comment (kripken) [inline docs/emcc.txt:433]: `use => using the`

File: docs/emcc.txt [hunk 4]
```diff
@@ -537,10 +537,11 @@ Options that are modified or new in *emcc* are listed below:
    multithreaded builds (-s USE_PTHREADS=1/2).
 
 "--em-config"
-   Specifies the location of the **.emscripten** configuration file
-   for the current compiler run. If not specified, the environment
-   variable "EM_CONFIG" is first read for this location. If neither
-   are specified, the default location **~/.emscripten** is used.
+   Specifies the location of the **.emscripten** configuration file.
+   If not specified emscripten will search for ".emscripten" first in
+   the emscripten dirctory itself, and then in the users home
```
Comment (kripken) [inline docs/emcc.txt:542]: `users => user's`

File: docs/process.md [hunk 5]
```diff
@@ -199,18 +199,22 @@ the wasm backend):
...
 key values in that file include:
 
  * `LLVM_ROOT`: The path to the LLVM binaries.
  * `BINARYEN_ROOT`: The path to binaryen (the binaries are expected in `/bin` under there; note that
     despite the name this differs from `LLVM_ROOT` which points directly to the binaries).
  * `NODE_JS`: The path to Node.js, which is needed internally.
 
+Shipping this config file inside the emscripten directory is the simplest
+way to ensure it won't interfere with any config file that user might have
+in thier home directory.
```
Comment (kripken) [inline docs/process.md:216]: typo in `their`

File: site/source/docs/tools_reference/emcc.rst [hunk 6]
```diff
@@ -351,8 +351,8 @@ Options that are modified or new in *emcc* are listed below:
   If using this in combination with ``--clear-cache``, be sure to specify
   this argument first.
 
-  The Emscripten cache defaults to being located in the path name stored
-  in the ``EM_CACHE`` environment variable or ``~/.emscripten_cache``.
+  The Emscripten cache defaults ``emscripten/cache`` but can be overridden
```
Comment (kripken) [inline site/source/docs/tools_reference/emcc.rst:354]: as before, `defaults => defaults to` and `use => using the`

File: site/source/docs/tools_reference/emcc.rst [hunk 7]
```diff
@@ -432,7 +432,10 @@ Options that are modified or new in *emcc* are listed below:
 .. _emcc-config:
 
 ``--em-config``
-  Specifies the location of the **.emscripten** configuration file for the current compiler run. If not specified, the environment variable ``EM_CONFIG`` is first read for this location. If neither are specified, the default location **~/.emscripten** is used.
+  Specifies the location of the **.emscripten** configuration file.  If not
+  specified emscripten will search for ``.emscripten`` first in the emscripten
+  dirctory itself, and then in the users home directory (``~/.emscripten``).
```
Comment (kripken) [inline site/source/docs/tools_reference/emcc.rst:437]: typo in `directory` and `users => user's`

File: site/source/docs/tools_reference/emsdk.rst [hunk 8]
```diff
@@ -81,7 +84,8 @@ Emscripten Compiler Configuration File (.emscripten)
 
 The *Compiler Configuration File* stores the :term:`active <Active Tool/SDK>` configuration on behalf of the *emsdk*. The active configuration defines the specific set of tools that are used by default if Emscripten in called on the :ref:`Emscripten Command Prompt <emcmdprompt>`.
 
-The configuration file is named **.emscripten**. It is user-specific, and is located in the user's home directory (**~/.emscripten** on Linux).
+The configuration file is named **.emscripten**. It is emsdk-specific, so it
+wont conflict with any config file the user might have in thier home directory.
```
Comment (kripken) [inline site/source/docs/tools_reference/emsdk.rst:88]: `wont => won't`

`thier => their`

File: tools/settings_template.py [hunk 9]
```diff
@@ -1,5 +1,5 @@
-# This file will be edited (the {{{ }}} things), and then ~/.emscripten created
-# with the result, if ~/.emscripten doesn't exist.
+# This file will be edited (the {{{ }}} things), and written to `.emscripten`
+# when emscripten is direct uses and no config file is found.
```
Comment (kripken) [inline tools/settings_template.py:2]: i don't understand this sentence. "direct uses" - what is "direct" referring to?

could this be `when emscripten is used and no config file is found` ?

**Comment by kripken** [comment]
Looks like this broke the chromium roll, https://logs.chromium.org/logs/emscripten-releases/buildbucket/cr-buildbucket.appspot.com/8878626006476971600/+/steps/Emscripten_testsuite__upstream__wasms_/0/stdout

**Comment by kripken** [comment]
Looks like now it prints
```
==============================================================================
Welcome to Emscripten!

This is the first time any of the Emscripten tools has been run.
```
When it didn't do so before.

**Comment by sbc100** [comment]
I did expect some fallout.. investigating now.

**Pull request merged**